### DISCUSSION TITLE
Will MAUI finally be _the_ .NET-GUI-Framework of choice?

### DISCUSSION THREAD

**Discussion by griestopf** [dotnet/maui#173] [General]
When I first heard about MAUI I thought "finally": MS starts cleaning up the .NET GUI-platform hell (WinForms, WPF, Xamarin Forms, ...). The only thing we know about the planned project progress so far is a first release date at the end of 2021 and that it will NOT support Linux

Obviously there has been no work on this repository for two months now. What are the planned Mile stones/project progress/next steps? Is this project still alive? Will MAUI be the answer to all the pain C#/.NET-Developers experience when looking for a decent platform-independent GUI-platform? Can I gamble on this horse for products shipped after 2021? 

Just to mention: I am working for a big research institute and we are currently evaluating different GUI technologies to host functionality written in C# (and some C++). The two top-of-the-list candidates are Qt(!) and Electron(!). None of the existing native C# Frameworks (including third-party-solutions) are nearly feasible.

**Comment by mrwcjoughin** [comment]
this is a brand new project and is and is only scheduled to be released with .NET 6 which is November 2021

https://github.com/xamarin/Xamarin.Forms/wiki/Feature-Roadmap#july-15-2020-pull-request-policy-changes <https://github.com/xamarin/Xamarin.Forms/wiki/Feature-Roadmap#july-15-2020-pull-request-policy-changes>

**Comment by ccarlo88** [comment]
> Just to mention: I am working for a big research institute and we are currently evaluating different GUI technologies to host functionality written in C# (and some C++). The two top-of-the-list candidates are Qt(!) and Electron(!). None of the existing native C# Frameworks (including third-party-solutions) are nearly feasible.

@griestopf you mentioned you are working at a research institute and I don't know what is your project, so I imagine this could be a complex desktop application. MAUI is just a platform for mobile applications, like Android & IOs solution. If you are targeting something complex, in .NET world I would recommend WPF. Yes, it's Windows only but it is as far as you can go in a complex application in the .NET environment.

**Comment by bmacombe** [comment]
MAUI plans to make Windows and Mac desktop first-class targets along with iOS and Android.

**Comment by GeraudFabien** [comment]
@griestopf @bmacombe 
>  it will NOT support Linux

> MAUI plans to make Windows and Mac desktop first-class targets along with iOS and Android.

I don't know how much time you pass on you're research but on the main readme of the repo there is a target list support 


Xamarin.Forms | .NET MAUI
-- | --
Platforms |   |  
Android | API 19+ | API 21+
iOS | 9-15 | 10+
Linux | Community | Community
macOS | Community | Microsoft
Tizen | Samsung | Samsung
Windows | UWP Microsoft WPF Community | Microsoft


There is linux support... In fact Xamarin support already [Linux](https://docs.microsoft.com/en-us/xamarin/xamarin-forms/platform/other/gtk?tabs=windows) and [Windows](https://docs.microsoft.com/en-us/xamarin/xamarin-forms/platform/other/wpf) (It's in preview so not production ready). Also UNO as port Xamarin to the Web. So if they successfully port MAUI too (since maui will be based on Xamarin code it may be possible) the support list should be

Xamarin.Forms | .NET MAUI
-- | --
Platforms |   |  
Android | API 19+ | API 21+
iOS | 9-15 | 10+
Linux | Community | Community
macOS | Community | Microsoft
Tizen | Samsung | Samsung
Windows | UWP Microsoft WPF Community | Microsoft
Web | UNO
 
For the web as first class solution it's discuss [here #62](https://github.com/dotnet/maui/issues/62)

**Reply by GeraudFabien** [reply]
Oups i forget the link to UNO.
[Docs](https://platform.uno/xamarin-forms/)
[Repo](https://github.com/unoplatform/Uno.Xamarin.Forms.Platform)

**Reply by trampster** [reply]
The current linux support in Xamarin.Forms will not work in .net MAUI because it relies on the old mono GTK#2 lib which doesn't work on .NET 6.

The linux support will have to be written from the ground up against the community GTK#3 library which supports .NET 6. So far no one from the community has stood up to do this work. And the last community contribution to Xamarin.Linux was over a year ago.

If you are choosing a GUI framework for a new project I wouldn't count on Maui working on Linux.

**Reply by griestopf** [reply]
Thank you. BTW: Not one single word about MAUI in Richard Landers .NET 6 Preview 1 announcement (https://devblogs.microsoft.com/dotnet/announcing-net-6-preview-1/).

**Reply by ccarlo88** [reply]
I hardly believe someone will in fact use MAUI. It's one of those projects which are dead before they were born.
By the way, can someone please bring Ballmer back?

**Reply by holmbuar** [reply]
> BTW: Not one single word about MAUI in Richard Landers .NET 6 Preview 1 announcement 

@griestopf
 Is [.NET Multi-platform App UI](https://devblogs.microsoft.com/dotnet/announcing-net-6-preview-1/#net-multi-platform-app-ui) not a section of the blog post you seem to be referring to?
